scheduler/planner.go

```diff
@@ -281,13 +260,74 @@ func (unit *Unit) RankValue() int64 {
 	// Increase the value for the number of dependencies, so that
 	// tasks (and units) which block other tasks run before tasks
 	// that don't block other tasks.
-	unit.cachedValue += priority * (numDeps / length)
+	value += priority * (u.NumDeps / length)
 
 	// Increase the value for tasks with longer runtimes, given
 	// that most of our workloads have different runtimes, and we
 	// don't want to have longer makespans if longer running tasks
 	// have to execute after shorter running tasks.
-	unit.cachedValue += priority * unit.distro.GetExpectedRuntimeFactor() * int64(math.Floor(expectedRuntime.Minutes()/float64(length)))
+	value += priority * u.Settings.GetExpectedRuntimeFactor() * int64(math.Floor(u.ExpectedRuntime.Minutes()/float64(length)))
+
+	return value
+}
+
+func (unit *Unit) info() unitInfo {
```

Comment:
No change, but I like the way this was rewritten.

Reply:
thank you!